web/lib/firebase/api.ts

```diff
@@ -143,6 +143,23 @@ export function createPost(params: {
   return call(getApiUrl('createpost'), 'POST', params)
 }
 
+export function boostMarket(params: any) {
+  return call(getApiUrl('boost-market'), 'POST', params)
+}
+
+let nonce = 0
+export function redeemBoost(params: any) {
+  // TODO: rate limit on the backend instead?
+  const now = Date.now()
+  if (now - nonce < 15 * 1000) {
+    throw Error(
+      'Please wait 15 seconds after redeeming a boost before redeeming another.'
+    )
```

Comment:
Not a great user experience...  Ok to try it for now.

I think instead of the button we might want auto-claim with 5 seconds of viewing the ad on screen (with countdown 5-4-3-2-1).